catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/SubscriptionsPersistentStoreImpl.java

```diff
@@ -0,0 +1,220 @@
+package org.codice.ddf.catalog.ui.query.monitor.impl;
+
+import static org.apache.commons.lang3.Validate.notBlank;
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.codice.ddf.catalog.ui.query.monitor.api.SubscriptionsPersistentStore;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Associate sets of emails with an ID. This implementation will preserve any other
+ * properties stored under the ID.
+ */
+public class SubscriptionsPersistentStoreImpl extends AbstractSubscriptionsPersistentStore {
```

Comment:
I think we are going to have thread safety issues if multiple people are trying to add and remove their email to the same id.